### ISSUE TITLE
Photos app in NC 24.0.0 RC1 using version Photos 1.6.1 keeps loading more thumbnails

### ISSUE THREAD

**Issue by valdearg** [nextcloud/photos#1081]
**Describe the bug**
When viewing the Photos tab, this normally loads the most recently modified images on the site. As you scroll down the page, more images are loaded up.

In the latest version of 24.0.0 RC1 I noticed that the page performance was extremely sluggish in comparisson to the previous 23 versions. 

After monitoring the page, it turns out this is because it's stuck in an infinite loop of requesting more images, making the page performance eventually poor.

Checking the Network tab in Developer Tools gives me a constant series of requests to the dav/ enpoint requesting the SEARCH metadata URL. 

**To Reproduce**
Steps to reproduce the behaviour:
1. Go to the Photos menu(?) along the top of the NC page.
2. Leave the page be and watch the scrollbar on the side of the page. This continues to grow as more images are requested.
3. Load up F12 developer tools and monitor the Network tab, the requests to the dav/ API occur about once every second.

**Expected behaviour**
I would expect this to request further images only when I scroll down, rather than constantly.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser: Chrome v100, also tried in Firefox.
 - Version: 24.0.0 RC1

**Comment by valdearg** [comment]
Anyone look at this? Still occurring in the release version. Is this repo active? Is there a better one to use?

**Comment by skjnldsv** [comment]
hey @Mikescops  :)
Anything changed in the virtual scrolling library?

**Comment by Mikescops** [comment]
Hey @skjnldsv, I made a few updates and bugfix here and there but nothing major, also I bumped version (0.X.0) to make sure it doesn't break anything on services using this dependency.

**Comment by Mikescops** [comment]
Wow indeed, I just made the update to v24 and it's looping fast, I can take a look

**Comment by Mikescops** [comment]
What I have for now:

1. First load: gridHeight: 0, elementWindowOffset: 94, we trigger the batch load
2. Second load: gridHeight: 2896, elementWindowOffset: 50, we skip the load

And then, there are plenty of calls with gridHeight: 0, elementWindowOffset: 0 which triggers the load of images.

Currently looking at why the gridHeight goes to 0

**Comment by Mikescops** [comment]
For some reason `element.getBoundingClientRect()` returns 0 for the height

I don't get why, is the page rerendered and I don't see it. At least on my demo project I don't have this issue.

**Comment by Mikescops** [comment]
Gosh that is odd


![Screenshot_20220510_130936](https://user-images.githubusercontent.com/4266283/167615566-ebfce1d1-c712-44a7-baf2-b21868cf66f9.png)


![Screenshot_20220510_130915](https://user-images.githubusercontent.com/4266283/167615567-5123945e-36d3-4c4a-b313-d4c6c401ab53.png)


![Screenshot_20220510_130825](https://user-images.githubusercontent.com/4266283/167615569-9c3437f8-6778-47e1-b44e-b8cfce2afb86.png)

**Comment by Mikescops** [comment]
At a moment, there is no more items in the list to render. 
I'm not expert with vue, but given that we're rebuilding the full item list at each batch load, is it possible that there is a frame/laps of time where the item list is empty during rerender?

**Comment by valdearg** [comment]
Looks like behaviour may have changed a little in 14.0.1.1, in Chrome and Firefox I get the following error:

`SW registration failed:  TypeError: Failed to register a ServiceWorker for scope ('https://<NC URL>/') with script ('https://<NC URL>/apps/photos/service-worker.js'): A bad HTTP response code (404) was received when fetching the script.`

![image](https://user-images.githubusercontent.com/2789033/170874158-e5fcab5e-8090-40ea-8a20-49bd570e3ce1.png)

It sometimes loads more images and sometimes just does nothing, eventually getting an error message which is just "an error occurred", I can't seem to find out exactly what triggers the error, other than that this shows:

```Error fetching timeline Error: Request failed with status code 504
    at e.exports (createError.js:16:15)
    at e.exports (settle.js:17:12)
    at XMLHttpRequest.E (xhr.js:66:7)
```
In Timeline.vue:281

Firefox seems to perform a bit better, it doesn't make the frequent requests to the remote.php/dav/ endpoint.

![image](https://user-images.githubusercontent.com/2789033/170874701-0f70c7a2-6c35-4597-9eff-a7e9c54a1d5b.png)

**Comment by marcelklehr** [comment]
This is likely due to me plowing through the code. Sorry for not seeing this issue sooner. Here's the fix: #1112

**Comment by Mikescops** [comment]
Oh lol, it was just this line :man_facepalming: 
@marcelklehr this one got me crazy and i dropped it ahah, but now it makes lot of sense!
